clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java

```diff
@@ -38,17 +39,19 @@
     public static final String ZK_SASL_CLIENT = "zookeeper.sasl.client";
     public static final String ZK_LOGIN_CONTEXT_NAME_KEY = "zookeeper.sasl.clientconfig";
 
+    private JaasUtils() {}
+
     /**
      * Construct a JAAS configuration object per kafka jaas configuration file
      * @param loginContextName
      * @param key
      * @return JAAS configuration object
      */
-    public static String jaasConfig(String loginContextName, String key) throws IOException {
+    public static String jaasConfig(String loginContextName, String key) {
         AppConfigurationEntry[] configurationEntries = Configuration.getConfiguration().getAppConfigurationEntry(loginContextName);
         if (configurationEntries == null) {
-            String errorMessage = "Could not find a '" + loginContextName + "' entry in this configuration.";
-            throw new IOException(errorMessage);
+            LOG.warn("Could not find a '{}' entry in this configuration.", loginContextName);
+            return null;
```

Comment:
any reason to not throw exception here. This particular method when user specifically states that they want to use SASL in that case throwing an exception is better option no?